src/web/modules/gateway_accounts/gateway_accounts.http.ts

```diff
@@ -152,17 +152,23 @@ const writeAccount = async function writeAccount(req: Request, res: Response): P
 
   logger.info(`Created new Gateway Account ${gatewayAccountIdDerived}`)
 
-  // connect system linked services to the created account
+  // connect system linked services to the created account  
   if (linkedService) {
     await AdminUsers.updateServiceGatewayAccount(
       linkedService,
       gatewayAccountIdDerived
     )
-    logger.info(`Service ${linkedService} linked to new Gateway Account ${gatewayAccountIdDerived}`)
+    logger.info(`Service ${linkedService} linked to new Gateway Account ${gatewayAccountIdDerived}`)    //linkedService.current_go_live_stage !== LIVE
+    
+    const serviceDetails = await AdminUsers.service(linkedService)
+    const isServiceAlreadyLive = serviceDetails.current_go_live_stage === 'LIVE'
 
     if (account.isLive()) {
-      await AdminUsers.updateServiceGoLiveStatus(linkedService, 'LIVE')
-      logger.info(`Service ${linkedService} 'current_go_live_stage' updated to 'LIVE'`)
+      await AdminUsers.updateServiceDetails(linkedService, account.isLive(), isServiceAlreadyLive, account.sector, account.internalFlag)
```

Comment:
because we're doing a patch either way, looks like we can dispense with the if/else and just call `updateServiceDetails` once, passing in the relevant parameters

Reply:
I am thinking about keeping this.  As I thought it could get confusing with too many parameters and also building up the logger message. Let me know what you think.